### ISSUE TITLE
registerJs, registerCss (etc) after View::endBody()

### ISSUE THREAD

**Issue by atiline** [yiisoft/yii2#18328]
Hi!

I suggest throwing an exception or calling \Yii::warning or \Yii::error when functions yii\web\View::registerJs, yii\web\View::registerCss etc are called after function yii\web\View::endBody() was called.

Now there are any checks of Application or View states.

Some example - you can call registerJs() after Application::EVENT_AFTER_REQUEST. Any JS will be added to response.

**Comment by samdark** [comment]
Sounds alright. But may I wonder how did you experience that?

**Comment by lubosdz** [comment]
Formarly sounds reasonable but applies limits to those rare justified cases I've seen around.
So this would be either an error or an intention of a developer - in first case framework should not apply BC breaking limits, in second case framework should not attempt to solve developers mistake. It's edge case.

**Comment by samdark** [comment]
I wonder if registering anything after request could be a valid use-case...

@yiisoft/reviewers, @yiisoft/core-developers, @yiisoft/yii2 what do you think?

**Comment by bizley** [comment]
I agree with @lubosdz - even though it is a valid issue it's not common and should be resolved by the developer.

**Comment by My6UoT9** [comment]
I think the wish of the OP is not to support this behaviour, but to throw some warning when this happens.
Just throwing a Notice should not be a BC?

**Comment by bizley** [comment]
Yes, warning could be added, indeed.
